hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieDataBlock.java

```diff
@@ -79,12 +78,11 @@ public abstract class HoodieDataBlock extends HoodieLogBlock {
    * NOTE: This ctor is used on the write-path (ie when records ought to be written into the log)
    */
   public HoodieDataBlock(List<HoodieRecord> records,
-                         boolean shouldWriteRecordPositions,
                          Map<HeaderMetadataType, String> header,
                          Map<FooterMetadataType, String> footer,
                          String keyFieldName) {
     super(header, footer, Option.empty(), Option.empty(), null, false);
-    if (shouldWriteRecordPositions) {
+    if (containsBaseFileInstantTimeOfPositions()) {
```

Comment:
we could make `supportWritePosition" as a method to HoodieIndex and avoid sorting the records . 
for bloom, simple etc we can blindly go ahead to write record positions. 
but other indexes, we can skip.

and so, we can avoid sorting the records too.

Reply:
alternatively, we could just check first record for valid position. 
if yes, we can sort and write the header. 
if not, we can bail out. no need to incur the overhead due to sorting

Reply:
Fixed.  Also added `HoodieLogBlock#addRecordPositionsIfRequired` to avoid code duplication.